src/libraries/System.Globalization/tests/Invariant/InvariantMode.cs

```diff
@@ -1206,6 +1206,41 @@ public void TestCultureInfo_Ctor_Int32_ReturnsInvariant(int culture)
             Assert.Equal(new CultureInfo(culture), CultureInfo.InvariantCulture);
         }
 
+        [Fact]
+        public void TestCasing()
+        {
+            // This test verifies that the casing table in the invariant globalization mode.
+            // The casing table is what we generate from the Unicode data files, and implemented in the CharUnicodeInfo.cs file.
+
+            TextInfo textInfo = CultureInfo.InvariantCulture.TextInfo;
+
+            for (int i = 0; i <= 0xFFFF; i++)
```

Comment:
As this is UNICODE-16, we should probably should keyhole-out:

- [Unicode BMP Private Use Area](https://en.wikipedia.org/wiki/Private_Use_Areas) U+E000..U+F8FF 
- [Unicode Upper and Lower surrogates ranges](https://en.wikipedia.org/wiki/Universal_Character_Set_characters#Surrogates) U+D800..U+DFFF

Thus the top of the loop should be
```csharp
for (int i = 0; i <= 0xFFFF; i++)
{
   if (i >= 0xD800 && i <= 0xF8FF)
      continue;
```

The surrogates should always travel in groups, and the PUA is really private use, and thus shouldn't be taunted.

Reply:
Although this doesn't really matter for this test anyway, I'll skip surrogates in the check.